WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -48,6 +51,9 @@ def create
         @competition.errors[:competition_id_to_clone] = "invalid"
       end
     end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates << current_user
+    end
 
     if @competition.errors.size == 0 && @competition.save
```

Comment:
If you don't call `valid?` (or `save`) on `@competition` first, `@competition.errors.size` will always be `0`.